server/routerlicious/packages/services-utils/src/scheduleJob.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import * as cron from "node-cron";
+import { ILogger } from "@fluidframework/server-services-core";
+import { Provider } from "nconf";
+
+export async function scheduleJob<T>(
+    api: () => Promise<T>,
+    config: Provider,
+    featureGateConfigName: string,
+    callName: string,
+    cronExpressionConfigName: string,
```

Comment:
Is passing the config and config flag names necessary? Might be more concise and future-proof to let the caller handle the how/when to call this, without making the implementation of this function dependent on nconf. Maybe personal pref, but
```ts
const deletionJobEnabled: boolean = config.get("deleteJob:enabled");
const deletionJobCronExpression: string = config.get("deleteJob:cronExpression");
if (deletionJobEnabled &&  deletionJobCronExpression) {
  scheduleJob(deletionJobFunction, "callname", deletionJobCronExpression);
}
```
seems more extensible/future-proof than
```ts
scheduleJob(deletionJobFunction, config, "deleteJob:enabled", "callname" "deleteJob:cronExpression");
```

Although I can see how the 2nd example shows that usage with the current implementation is more concise, the reality is that all that code is just in the function implementation instead, and locks the function into always having to be initialized with a config. Maybe in the future we want to just use this function as a standalone, or we don't need a feature gate for it, or we want to write unit tests without the overhead of nconf. In that case, we could do a very simple
```ts
scheduleJob(deletionJobFunction, "callname", "cron-expression-idk-what-this-looks-like")
```

Reply:
I just made a similar comment below before I saw this. Agree with @znewton

Reply:
Sorry for the late reply, that makes sense, will fix!